core/frontend/src/IModelConnection.ts

```diff
@@ -887,14 +892,134 @@ export namespace IModelConnection {
      * why the extents could not be obtained (e.g., because the Id did not identify a [GeometricModel]($backend)).
      */
     public async queryExtents(modelIds: Id64String | Id64String[]): Promise<ModelExtentsProps[]> {
+      const modelExtentsQuery = `
+      SELECT
+        Model.Id AS ECInstanceId,
+        iModel_bbox_union(
+          iModel_placement_aabb(
+            iModel_placement(
+              iModel_point(Origin.X, Origin.Y, 0),
+              iModel_angles(Rotation, 0, 0),
+              iModel_bbox(
+                BBoxLow.X, BBoxLow.Y, -1,
+                BBoxHigh.X, BBoxHigh.Y, 1
+              )
+            )
+          )
+        ) AS bbox
+      FROM bis.GeometricElement2d
+      WHERE InVirtualSet(:ids64, Model.Id) AND Origin.X IS NOT NULL
+      GROUP BY Model.Id
+      UNION
+      SELECT
+        ge.Model.Id AS ECInstanceId,
+        iModel_bbox(
+          min(i.MinX), min(i.MinY), min(i.MinZ),
+          max(i.MaxX), max(i.MaxY), max(i.MaxZ)
+        ) AS bbox
+      FROM bis.SpatialIndex AS i, bis.GeometricElement3d AS ge, bis.GeometricModel3d AS gm
+      WHERE InVirtualSet(:ids64, ge.Model.Id) AND ge.ECInstanceId=i.ECInstanceId AND InVirtualSet(:ids64, gm.ECInstanceId) AND gm.$->isNotSpatiallyLocated?=false
+      GROUP BY ge.Model.Id
+      UNION
+      SELECT
+        ge.Model.Id AS ECInstanceId,
+        iModel_bbox_union(
+          iModel_placement_aabb(
+            iModel_placement(
+              iModel_point(ge.Origin.X, ge.Origin.Y, ge.Origin.Z),
+              iModel_angles(ge.Yaw, ge.Pitch, ge.Roll),
+              iModel_bbox(
+                ge.BBoxLow.X, ge.BBoxLow.Y, ge.BBoxLow.Z,
+                ge.BBoxHigh.X, ge.BBoxHigh.Y, ge.BBoxHigh.Z
+              )
+            )
+          )
+        ) AS bbox
+      FROM bis.GeometricElement3d AS ge, bis.GeometricModel3d as gm
+      WHERE InVirtualSet(:ids64, ge.Model.Id) AND ge.Origin.X IS NOT NULL AND InVirtualSet(:ids64, gm.ECInstanceId) AND (gm.$->isNotSpatiallyLocated?=true OR gm.$->isNotSpatiallyLocated? IS NULL)
```

Comment:
I believe, our assumption here may have been wrong. When `isNotSpatiallyLocated` is `NULL`, it should be treated as if value is `false` and not `true`